// mpackage/main.js
exports.myPackage = ({name,dateBirth, purpose}) => {
    const {firstName,lastName} = separateAndCapitalize(name)

    return {
        id: generateRandomId(), 
        firstName,
        lastName,
        dateBirth,
        age: calculateAge(dateBirth),
        purpose: capitalizedFirstLetter(purpose), 
      }
}
    

function calculateAge(dateBirth) {
  const today = new Date();
  const birthDate = new Date(dateBirth);
  let age = today.getFullYear() - birthDate.getFullYear();
  const month = today.getMonth() - birthDate.getMonth();
  if (month < 0 || (month === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
}

function capitalizedFirstLetter(data = "") {
  return data.charAt(0).toUpperCase() + data.slice(1).toLowerCase();
}


function generateRandomId() {
  const characters = "abcdefghijklmnopqrstuvwxyz0123456789";
  let id = "";
  for (let i = 0; i < 10; i++) {
    const randomIndex = Math.floor(Math.random() * characters.length);
    id += characters.charAt(randomIndex);
  }
  return id;
}

function separateAndCapitalize(fullName) {
  const parts = fullName.split(" ");

  const firstName = parts?.[0];
  const surname = parts?.[1];

  const capitalizedFirstName = firstName
    ? capitalizedFirstLetter(firstName)
    : "";
  const capitalizedSurname = surname ? capitalizedFirstLetter(surname) : "";

  return { firstName: capitalizedFirstName, lastName: capitalizedSurname };
}
